### DISCUSSION TITLE
All Templates Script

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1829] [Announcements]
A script designed to allow for the creation of one of the many free LXC templates. Great for creating system LXCs. 

The script creates a `*.creds` file in the Proxmox root directory with the password of the newly created LXC.

Please take note that if you plan to use this script for creating TurnKey LXCs, you'll need to modify the hostname after creation.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/all-templates.sh)"
```

⚡ Default Settings: 2GB RAM - 8GB Storage - 2vCPU - onboot 0 - DHCP - Unprivileged⚡

Resource and network settings are adjustable post LXC creation.

**Comment by tteck** [comment]
Users can download the script, customize it as necessary, and then utilize it to create any template container from a single script.

Download (all executed within the Proxmox shell):
```
wget -q https://github.com/tteck/Proxmox/raw/main/misc/all-templates.sh
```
Edit:
```
nano all-templates.sh
```
Execute:
```
bash all-templates.sh
```

**Comment by connectivityengineer** [comment]
This is awesome 
It looks like you have removed the ability to pick the container number and set the ip address. 

We generally set the container name by matching it to the last 2 octets of the ip address 

I am not sure where or how to add that functionality back into the script - even so this is great work. 

We had to manually set the ip for now in the script - because it does not have that funcationality and we do not have a DHCP server running. 

I would love to know if you are considering adding the functionality back to the all templates script.

**Reply by tteck** [reply]
At this time, there are no imminent plans to enhance the script's functionality.

**Comment by tteck** [comment]
Here's a set of customizable options that can be tailored to better align with your specific system requirements:


![Screenshot 2023-11-01 10 23 15 AM](https://github.com/tteck/Proxmox/assets/21342670/b97af931-3524-41f5-99e5-6bab18eccf9c)
**_Note: GATEWAY is required if Static IPv4 CIDR Address is used._**
